### ISSUE TITLE
Question: (functionality)

### ISSUE THREAD

**Issue by derei** [MrRar/gps_locker#4]
How does it impact battery? Have you done any testing in this regard?
Also, what happend in situations when the gps can't get a lock (no signal)?

**Comment by MrRar** [comment]
> How does it impact battery? Have you done any testing in this regard?

I have done no testing. I presume this app would be detrimental to the battery. Pulling the GPS every second: I can't imagine that would be good for battery.

> Also, what happend in situations when the gps can't get a lock (no signal)?

It will just keep trying forever.

**Comment by derei** [comment]
Wouldn't you say that both issues should be addressed?
Perhaps if the app can monitor the connection, how many satellites are used, and if their number drops under a certain value, to call again, for example.

As for senseless trying, it should try for some time, and then pause until something changes (phone moves for a certain distance, for example). The movement can be estimated with accelerometer sensors too, if gps is not yet locked.

In conclusion, it definitely requires some smart management, to protect battery.

**Comment by MrRar** [comment]
The main goal of the app is obviously to keep the GPS locked as much as possible. Doing anything that might compromise that goes against the goal. For example, pulling the GPS less frequently if it has a bad signal. In this case it would be up to the user to turn off the app if they want to save battery.

**Comment by MrRar** [comment]
I'm not apposed to saving battery. I just don't know how to do it without compromising the main goal of the app.
